lib/mix/lib/mix/compilers/test.ex

```diff
@@ -45,17 +45,19 @@ defmodule Mix.Compilers.Test do
         warnings_as_errors? = Keyword.get(opts, :warnings_as_errors, false)
 
         try do
-          failed? =
+          compilation_failed? =
             case Kernel.ParallelCompiler.require(test_files, parallel_require_callbacks) do
               {:ok, _, [_ | _]} when warnings_as_errors? -> true
               {:ok, _, _} -> false
               {:error, _, _} -> exit({:shutdown, 1})
             end
 
+          :ok = ExUnit.configure(skip_failures_manifest_file: compilation_failed?)
```

Comment:
Unfortunately there is a race condition here. Maybe the tests finished before we set this option.

I am thinking for now the simplest option may be to get rid of the failed manifest altogether indeed. :( Otherwise we will need to expose full control of the manifest via ExUnit APIs.

Reply:
So we're gonna need something like ExUnit.clear_failed_manifest() to call after everything stops?

Btw, what's the actual race condition?

Reply:
Oh, because await_run wait for the tests to finish, but they are started above?

Could we move the ParallelCompiler code to before the task is started? This should enable this idea to work

Reply:
> Btw, what's the actual race condition?

The race condition is that the tests finish and write to the manifest file before this call happens. :)

> So we're gonna need something like ExUnit.clear_failed_manifest() to call after everything stops?

I think we pass the manifest file to ExUnit, so we can just go ahead and delete it if it is available.

Reply:
> Could we move the ParallelCompiler code to before the task is started? This should enable this idea to work

Then we no longer load and run tests concurrently, it will make the suite slower.

Reply:
Yes. IIRC it resides in the :failed_manifest_file option

Reply:
What if we set the files with warnings as failed in the manifest, instead of deleting it?

Reply:
I like that! The best option so far and it should be just one small API addition.